src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/HistoricalDataQueueManager.java

```diff
@@ -71,11 +74,31 @@ public static synchronized long getQueueFifoTimeSpanMins() {
    * Set the FIFO queue time span in minutes.
    *
    * @param timeSpanMins new FIFO queue time span in minutes
+   * @throws IllegalArgumentException if unable to read a file
    */
   public static synchronized void setQueueFifoTimeSpanMins(long timeSpanMins) {
+    if (timeSpanMins < HistoricalDataConstants.MIN_QUEUE_SPAN_MINS) {
+      throw new IllegalArgumentException(
+          "The FifoTimeSpan must not be less than "
+              + HistoricalDataConstants.MIN_QUEUE_SPAN_MINS
+              + ".");
+    }
     queueFifoTimeSpanMins = timeSpanMins;
   }
 
+  /**
+   * Set the maximum amount the Historical FIFO queue can get behind in minutes.
+   *
+   * @param timeMins new FIFO queue time span in minutes
+   * @throws IllegalArgumentException if parameter is not greater than 0
+   */
+  public static synchronized void setQueueMaxBehindMins(long timeMins) {
+    if (timeMins < 1) {
+      throw new IllegalArgumentException("The parameter passed must be greater than 0.");
```

Comment:
```suggestion
   * @throws IllegalArgumentException if parameter is not greater than or equal to 1
   */
  public static synchronized void setQueueMaxBehindMins(long timeMins) {
    if (timeMins < 1) {
      throw new IllegalArgumentException("The parameter passed must be greater than or equal to 1.");
```
or
```suggestion
   * @throws IllegalArgumentException if parameter is not greater than 0
   */
  public static synchronized void setQueueMaxBehindMins(long timeMins) {
    if (timeMins <= 0) {
      throw new IllegalArgumentException("The parameter passed must be greater than 0.");
```

Reply:
ok... this is like Sergio's 1+1 != 2

Reply:
Yeah, there is nothing technically wrong with how you had it. Its just better code style to match the value in the log and comments to the value being used in the functional portion of the code.

Reply:
You could even have this which would make it so that any change to the value would require just one line change. 
```suggestion
   * @throws IllegalArgumentException if timeMin parameter is invalid
   */
  public static synchronized void setQueueMaxBehindMins(long timeMins) {
    int minimumTimeMin = 0;
    if (timeMins <= minimumTimeMin) {
      throw new IllegalArgumentException("The parameter passed must be greater than " + minimumTimeMin + ".");
```

Reply:
> You could even have this which would make it so that any change to the value would require just one line change.

While this simplifies the code slightly and is less work if the value needs to change, I think it is much less user friendly. Users' IDEs may suggest Javadocs to them as they develop, and by removing the number from the Javadocs, the developer either has to guess on the constraint and hope they don't get an exception or dig through our source code. For example, this is what I can see in IntelliJ for the parseHistoricalFile() method.

![image](https://user-images.githubusercontent.com/51409174/152603464-4664f7bd-8019-4650-b4bb-cc3d26dcef73.png)

Reply:
Probably would be better to just make it final and use @link then.